### PR TITLE
cardano-tracer: OpenMetrics compliance for Prometheus; fix `forHuman` output in journald

### PR DESCRIPTION
**Pull request by mgmeier**
# Description

This PR is mainly a rewrite of rendering the Prometheus metric exposition in `cardano-tracer`:
* For the response, directly fold a `Builder` through the `Store` sample instead of manipulating `Text` values.
* Improved OpenMetrics compliance of Prometheus exposition; also addresses #5140 .
* Prometheus help annotations can be provided via the new optional config value `metricsHelp`.

Additionally, it:
* fixes forwarding `forHuman` output to journald.
* sets the period to invalidate a memoized routing table to 5s.
* adds JSON response to the EKG server (same as Prometheus).

It futher affects:
* `cardano-node` / `trace-dispatcher`: The `trace-documentation` CLI command has now an optional arg `--output-metric-help` for creating machine-readable JSON output of metrics docs.
* `workbench`: it adds a default (yet incomplete) metrics help dictionary to the cardano-tracer nix service config.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Any changes are noted in the `CHANGELOG.md` for affected package
- [x] The version bounds in `.cabal` files are updated
- [X] CI passes. See note on CI.  The following CI checks are required:
  - [X] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [X] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [X] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-tracer/src/Cardano/Tracer/Handlers/Metrics/Utils.hs [hunk 1]
```diff
@@ -1,63 +1,42 @@
...
-  ( MetricName
-  , MetricValue
-  , MetricsList
-  , RouteDictionary(..)
-  , getListOfMetrics
-  , renderListOfConnectedNodes
-  , renderJson
-  , nodeNames
-  , computeRoutes
+  ( module Cardano.Tracer.Handlers.Metrics.Utils
```
Comment (jutaro) [inline cardano-tracer/src/Cardano/Tracer/Handlers/Metrics/Utils.hs:6]: I have never seen this! What is the meaning of this?

File: cardano-tracer/src/Cardano/Tracer/Handlers/RTView/Utils.hs [hunk 2]
```diff
@@ -1,23 +1,27 @@
 {-# LANGUAGE NamedFieldPuns #-}
 
 module Cardano.Tracer.Handlers.RTView.Utils
-  ( forAcceptedMetrics_
-  , forAcceptedMetricsUI_
-  , forConnected
-  , forConnected_
-  , forConnectedUI
-  , forConnectedUI_
+  ( module Cardano.Tracer.Handlers.RTView.Utils
```
Comment (jutaro) [inline cardano-tracer/src/Cardano/Tracer/Handlers/RTView/Utils.hs:4]: You like this crazy stuff

**Review by jutaro — APPROVED** [review]
LGTM

File: cardano-tracer/src/Cardano/Tracer/Handlers/Metrics/Utils.hs (hunk 1 above)
Comment (Icelandjack) [reply cardano-tracer/src/Cardano/Tracer/Handlers/Metrics/Utils.hs:6]: It re-exports: https://stackoverflow.com/questions/36044041/haskell-module-inside-module

Comment (mgmeier) [reply cardano-tracer/src/Cardano/Tracer/Handlers/Metrics/Utils.hs:6]: This exports all symbols from that module. It's best to use the current module name here when explicit export lists are required, and you want to export *everything*.

File: cardano-tracer/src/Cardano/Tracer/Handlers/RTView/Utils.hs (hunk 2 above)
Comment (mgmeier) [reply cardano-tracer/src/Cardano/Tracer/Handlers/RTView/Utils.hs:4]: Yes, I want to be explicit about: this module exports *every* definition and type it contains :)

**Review by Icelandjack — APPROVED** [review]
For MetricsHelp it suffices to use generalized newtype deriving but it doesn't hurt to be explicit. LGTM!

**Comment by mgmeier** [comment]
> For MetricsHelp it suffices to use generalized newtype deriving but it doesn't hurt to be explicit. LGTM!

GND serializes as such: `{"Left":"testfile.json"}`. I'm trying to avoid any Haskell / aeson specific serializations. Instead, provide one parser allowing for two distinct JSON types without relying on serializing a constructor or tag.

**Pull request merged**